doc/source/ci.rst

```diff
@@ -144,7 +144,54 @@ Jenkins Pipeline
 
     }
 
+The following `Jenkinsfile` uses the official 'quay.io/ansible/molecule' image.
 
+.. code-block:: groovy
+
+    pipeline {
+      agent {
+        docker {
+          image 'quay.io/ansible/molecule'
+          args '-v /var/run/docker.sock:/var/run/docker.sock'
+        }
+      }
+
+      stages {
+
+        stage ('Display versions') {
+          steps {
+            sh '''
+              docker -v
+              python -V
+              ansible --version
+              molecule --version
+            '''
+          }
+        }
+
+        stage ('Molecule test') {
+          steps {
+            sh 'sudo molecule test --all'
```

Comment:
TBH, my hopes were to see an example of spreding molecule execution in multiple CI stages, as this would be the only way to make it clear when it fails on certain platform.
 
Stages can run serially or in parallel, this being based on the job defintion but I would really like to see how to split a single execution in multiple ones.

Reply:
Something to be aware of is that the general convention as show in the docs, with Docker, is using instance names such as instance-1 and instance-2. If you were to run scenarios in parallel you must be careful of the following:

- The instances are named uniquely if your Jenkins executors are running on the same server. Otherwise the destroy phase would be run end a test run early, or a an instance could fail to startup due to a naming conflict.
- If you happened to run two branches of the same role/molecule repository then they could also conflict, so I'd recommend naming the instance after the current branch as well.

Using the following Jenkinsfile works for the later issue:

```
pipeline {
	agent any
	environment {
		BRANCH_SAFE = sh (script: "echo $BRANCH_NAME | sed 's@/@-@'", returnStdout: true).trim()
	}
	stages {
		stage ('test') {
			steps {
				sh 'molecule test --all'
			}
		}
	}
	post {
		always {
			deleteDir()
		}
	}
}
```

With molecule.yml looking like:
```
platforms:
  - name: ${BRANCH_SAFE:-default}-instance-1
    image: centos:latest
```

Obviously it could be expanded out a little to account for any scenario, e.g. by adding the a variable containing the current molecule scenario, but I didn't need that for my means.

Worth noting that this wouldn't be needed for certain setups of Jenkins where each executor has a completely isolated Docker server, but alas this isn't an option for me.

Reply:
So I agree about additional examples - and this can probably be worked out - but they become more advanced. My goal in publishing the simple example was to get it working for the multi-branch pipeline from the pre-built molecule image. 

The example above it, for me, did not work - it failed with errors on "pipeline durability". To get it to work, we would have had to remove alot of security from Jenkins (do you really want someone else's build script to install software on your slaves ?)

Based on the bug / bug fixes - I don't think I was the only one struggling with this.

Reply:
@ssbarnea 

I did the following and it works (did not place it in the PR yet. See bellow for other ideas I'm looking at.

```
pipeline {
  agent {
    docker {
      image 'quay.io/ansible/molecule'
      args '-v /var/run/docker.sock:/var/run/docker.sock'
    }
  }

  stages {

    stage ('Display versions') {
      steps {
        sh '''
          docker -v
          python -V
          ansible --version
          molecule --version
        '''
      }
    }

    stage ('Molecule pre-clean') {
      steps {
        sh 'sudo molecule destroy'
      }
    }

    stage ('Molecule converge') {
      steps {
        sh 'sudo molecule converge'
      }
    }

    stage ('Molecule idempotence') {
      steps {
        sh 'sudo molecule idempotence'
      }
    }

    stage ('Molecule apply side effect') {
      steps {
        sh 'sudo molecule side-effect'
      }
    }

    stage ('Molecule verify') {
      steps {
        sh 'sudo molecule verify'
      }
    }

    stage ('Molecule post-clean') {
      steps {
        sh 'sudo molecule destroy'
      }
    }

  } // close stages
}   // close pipeline
```

I would like to see If I can run the scenarios in parallel and via "dynamic discovery".